ios/browser/brave_ads/ads_service_impl_ios.mm

```diff
@@ -216,6 +217,16 @@
   NOTIMPLEMENTED() << "Not used on iOS.";
 }
 
+void AdsServiceImplIOS::ParseAndSaveCreativeNewTabPageAds(
+    const std::string& json,
+    ParseAndSaveCreativeNewTabPageAdsCallback callback) {
+  if (!IsInitialized()) {
+    return std::move(callback).Run(/*success*/ false);
+  }
+
+  ads_->ParseAndSaveCreativeNewTabPageAds(json, std::move(callback));
```

Comment:
Is this done on a task runner/off main? Don't really want to be doing JSON parsing on main unless we know the content is short

Reply:
This is small JSON which comes from CRX package. I've refactored json string to base::Value to avoid parsing in ads library

Reply:
Resolving comment since we’re splitting up the PR, and this will be addressed in a separate one. Thanks!